### PR TITLE
Improve exports, migrate project to `module: NodeNext`

### PR DESCRIPTION
**Pull request by Mati365**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Internal: Migrate project from `module: ESNext` to `module: NodeNext` to enforce `.js` import extensions.
Internal: Move reusable `test-utils` functions to `src/`.
Fix: Improved support for TS configurations containing `skipLibCheck: false` and non-installed `ckeditor5-premium-features`.

### CODE REVIEW

File: vite.config.ts [hunk 1]
```diff
@@ -13,20 +13,35 @@ export default defineConfig( {
 	build: {
 		target: 'esnext',
 		lib: {
-			entry: 'src/index',
-			fileName: 'index',
-			formats: [ 'es', 'umd' ],
-			name: 'CKEDITOR_INTEGRATIONS_COMMON'
+			formats: [ 'es', 'cjs' ],
```
Comment (filipsobol) [inline vite.config.ts:16]: This package needs a UMD build, as it's required by UMD builds of Vue and React integrations

File: src/test-utils.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import './globals.d';
```
Comment (filipsobol) [inline src/test-utils.ts:6]:
```suggestion
import './globals.js';
```

I'm not 100% sure about this

File: vite.config.ts (hunk 1 above)
Comment (Mati365) [reply vite.config.ts:16]: It's not possible to compile UMD multiple entry builds in Vite at this moment. I checked and it used to work normally.

File: src/test-utils.ts (hunk 2 above)
Comment (Mati365) [reply src/test-utils.ts:6]: It's definition file, with `declare module`, so it cannot be compiled to `.js` file. Compiler complains about that.

```tsx
⋊> ~/P/c/ckeditor5-integrations-common on ck/fix-extensions ⨯ yarn run build                              09:46:16
yarn run v1.22.19
$ vite build
vite v5.4.1 building for production...
✓ 2 modules transformed.
x Build failed in 2.70s
error during build:
Could not resolve "./globals.js" from "src/index.ts"
file: /home/mateusz/Projekty/cksource/ckeditor5-integrations-common/src/index.ts
```

File: vite.config.ts (hunk 1 above)
Comment (filipsobol) [reply vite.config.ts:16]: A `require` call will not work in a browser, so it's not a suitable substitute. We've made a decision (good or bad) to have UMD builds of Vue and React integrations, and thus we need UMD builds for dependencies too.

Vite exposes Rollup config that should allow for that.

**Review by filipsobol — DISMISSED** [review]

**Review by filipsobol — APPROVED** [review]

**Pull request merged**